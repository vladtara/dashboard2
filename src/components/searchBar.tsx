import React, { useEffect, useState } from "react";
import styled from "styled-components";

import { Button } from "./elements";

const Search = styled.form`
  width: 100%;
  height: 2rem;

  display: flex;

  padding-top: 0.25rem;
`;

const SearchInput = styled.input`
  width: 100%;
  margin: 0px;

  font-size: 1rem;

  border: none;
  border-bottom: 1px solid ${(props) => props.theme.accentColor};
  border-radius: 0;

  background: none;
  color: ${(props) => props.theme.mainColor};

  :focus {
    outline: none;
  }
`;

const SearchButton = styled(Button)`
  margin: 0px 2px;
  min-height: 0;
`;

export interface ISearchProviderProps {
  name: string;
  url: string;
  prefix: string;
}

export interface ISearchProps {
  placeholder: string;
  defaultProvider: string;
  autoFocus?: boolean;
  providers: Array<ISearchProviderProps> | undefined;
}

interface ISearchBarProps {
  search?: ISearchProps;
}

export const handleQueryWithProvider = (
  search: ISearchProps,
  query: string,
) => {
  const queryArray: Array<string> = query.split(" ");
  const prefix: string = queryArray[0];

  queryArray.shift();

  const searchQuery: string = queryArray.join(" ");

  let providerFound = false;
  if (search.providers) {
    search.providers.forEach((provider: ISearchProviderProps) => {
      if (provider.prefix === prefix) {
        providerFound = true;
        window.location.href = provider.url + searchQuery;
      }
    });
  }

  if (!providerFound) window.location.href = search.defaultProvider + query;
};

/**
 * Renders a search bar
 * @param {ISearchBarProps} search - The search providers for the search bar to use
 */
const SearchBar = ({ search }: ISearchBarProps) => {
  const [input, setInput] = useState<string>("");
  const [buttonsHidden, setButtonsHidden] = useState<boolean>(true);

  useEffect(() => setButtonsHidden(input === ""), [input]);

  if (search === undefined) return <></>;

  const handleSearchQuery = (e: React.FormEvent) => {
    const query: string = input || "";

    if (query.split(" ")[0].includes("/")) {
      handleQueryWithProvider(search, query);
    } else {
      window.location.href = search.defaultProvider + query;
    }

    e.preventDefault();
  };

  const autoFocus = () => {
    return search.autoFocus !== undefined ? search.autoFocus : false;
  };

  return (
    <Search onSubmit={(e) => handleSearchQuery(e)}>
      <SearchInput
        type="text"
        data-testid="search-input"
        value={input}
        placeholder={search.placeholder}
        autoFocus={autoFocus()}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
          setInput(e.target.value)
        }
      ></SearchInput>
      <SearchButton
        type="button"
        data-testid="search-clear"
        onClick={() => setInput("")}
        hidden={buttonsHidden}
      >
        Clear
      </SearchButton>
      <SearchButton
        type="submit"
        data-testid="search-submit"
        hidden={buttonsHidden}
      >
        Search
      </SearchButton>
    </Search>
  );
};

export default SearchBar;
