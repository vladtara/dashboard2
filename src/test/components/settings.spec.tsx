import { fireEvent, render } from "@testing-library/react";
import Settings, {
  FormContainer,
  Table,
  TableRow,
  TableCell,
  HeadCell,
  Section,
  SectionHeadline,
} from "../../components/settings";
import { ISearchProps } from "../../components/searchBar";
import { IThemeDataProps } from "../../lib/useTheme";

const themes: IThemeDataProps = {
  themes: [
    {
      label: "Classic",
      value: 0,
      mainColor: "#000000",
      accentColor: "#1e272e",
      backgroundColor: "#ffffff",
    },
    {
      label: "Classic",
      value: 1,
      mainColor: "#000000",
      accentColor: "#1e272e",
      backgroundColor: "#ffffff",
    },
  ],
};

const search: ISearchProps = {
  defaultProvider: "https://test.com?q=",
  placeholder: "",
  providers: [
    {
      name: "Allmusic",
      url: "https://www.allmusic.com/search/all/",
      prefix: "/a",
    },
    {
      name: "Discogs",
      url: "https://www.discogs.com/search/?q=",
      prefix: "/di",
    },
    {
      name: "Duck Duck Go",
      url: "https://duckduckgo.com/?q=",
      prefix: "/d",
    },
  ],
};

const propsList = [
  {
    themes: themes,
    search: search,
  },
  {
    themes: themes,
    search: undefined,
  },
  {
    themes: undefined,
    search: search,
  },
  {
    themes: undefined,
    search: undefined,
  },
];

describe("settings.tsx", () => {
  const location: Location = window.location;

  beforeEach(() => {
    // @ts-ignore
    delete window.location;

    window.location = {
      ...location,
      reload: jest.fn(),
    };
  });

  it("Tests forms", () => {
    const { baseElement } = render(<FormContainer />);
    expect(baseElement).toMatchSnapshot();
  });

  it("Tests tables", () => {
    const { baseElement } = render(
      <Table>
        <tbody>
          <TableRow>
            <HeadCell>Test</HeadCell>
          </TableRow>
          <TableRow>
            <TableCell>Test</TableCell>
          </TableRow>
        </tbody>
      </Table>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("Tests sections", () => {
    const { baseElement } = render(
      <Section>
        <SectionHeadline>Test</SectionHeadline>
      </Section>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("Tests settings rendering", () => {
    propsList.forEach((props) => {
      const { baseElement } = render(
        <Settings themes={props.themes} search={props.search} />,
      );

      expect(baseElement).toMatchSnapshot();
    });
  });

  it("Tests submit button", () => {
    const settings = render(<Settings themes={themes} search={search} />);

    fireEvent.click(settings.getByTestId("button-refresh"));
    expect(window.location.reload).toHaveBeenCalledTimes(1);
  });

  it("Tests light theme selection", () => {
    const settings = render(<Settings themes={themes} search={search} />);

    fireEvent.change(settings.getByTestId("select-light"), {
      target: { value: 0 },
    });

    fireEvent.click(settings.getByTestId("button-submit"));
    expect(window.location.reload).toHaveBeenCalledTimes(1);
  });

  it("Tests dark theme selection", () => {
    const settings = render(<Settings themes={themes} search={search} />);

    fireEvent.change(settings.getByTestId("select-dark"), {
      target: { value: 0 },
    });

    fireEvent.click(settings.getByTestId("button-submit"));
    expect(window.location.reload).toHaveBeenCalledTimes(1);
  });

  it("Tests theme selection", () => {
    const settings = render(<Settings themes={themes} search={search} />);

    fireEvent.change(settings.getByTestId("select-light"), {
      target: { value: 5 },
    });

    fireEvent.click(settings.getByTestId("button-submit"));
    expect(window.location.reload).toHaveBeenCalledTimes(0);
  });
});
