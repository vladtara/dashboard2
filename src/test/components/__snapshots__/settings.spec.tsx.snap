// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings.tsx Tests forms 1`] = `
<body>
  <div>
    <div
      class="sc-iqcoie mUaeo"
    />
  </div>
</body>
`;

exports[`settings.tsx Tests sections 1`] = `
<body>
  <div>
    <div
      class="sc-hAZoDl UZCWe"
    >
      <h3
        class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
      >
        Test
      </h3>
    </div>
  </div>
</body>
`;

exports[`settings.tsx Tests settings rendering 1`] = `
<body>
  <div>
    <div
      class="sc-papXJ dURgvU"
      hidden=""
    >
      <div
        class="sc-kDDrLX byPpqm"
      >
        <h2
          class="sc-dkzDqf jbDwiw"
        >
          Settings
        </h2>
        <button
          class="sc-iBkjds dRHSwq"
          data-testid="close-button"
        >
          <i
            class="sc-ftvSup cWSRlf"
          >
            close
          </i>
        </button>
      </div>
      <div
        class="sc-hHLeRK eelEQI"
      >
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Theme
          </h3>
          <div
            class="sc-iqcoie mUaeo"
          >
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Light
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-light"
              >
                <option
                  data-testid="option-light-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-light-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Dark
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-dark"
              >
                <option
                  data-testid="option-dark-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-dark-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-submit"
              >
                Apply
              </button>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-refresh"
              >
                Refresh
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Search Providers
          </h3>
          <p
            class="sc-fEOsli OFSxW"
          >
            Default Search Provider
          </p>
          <p
            class="sc-bjUoiL KndDd"
          >
            https://test.com?q=
          </p>
          <table
            class="sc-crXcEl BVLoO"
          >
            <tbody>
              <tr
                class="sc-evZas caYqCy"
              >
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Search Provider
                </th>
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Prefix
                </th>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Allmusic
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /a
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Discogs
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /di
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Duck Duck Go
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /d
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <button
      class="sc-iBkjds dRHSwq"
      data-testid="toggle-button"
    >
      <i
        class="sc-ftvSup cWSRlf"
      >
        settings
      </i>
    </button>
  </div>
</body>
`;

exports[`settings.tsx Tests settings rendering 2`] = `
<body>
  <div>
    <div
      class="sc-papXJ dURgvU"
      hidden=""
    >
      <div
        class="sc-kDDrLX byPpqm"
      >
        <h2
          class="sc-dkzDqf jbDwiw"
        >
          Settings
        </h2>
        <button
          class="sc-iBkjds dRHSwq"
          data-testid="close-button"
        >
          <i
            class="sc-ftvSup cWSRlf"
          >
            close
          </i>
        </button>
      </div>
      <div
        class="sc-hHLeRK eelEQI"
      >
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Theme
          </h3>
          <div
            class="sc-iqcoie mUaeo"
          >
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Light
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-light"
              >
                <option
                  data-testid="option-light-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-light-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Dark
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-dark"
              >
                <option
                  data-testid="option-dark-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-dark-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-submit"
              >
                Apply
              </button>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-refresh"
              >
                Refresh
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Search Providers
          </h3>
          <p
            class="sc-fEOsli OFSxW"
          >
            Default Search Provider
          </p>
          <p
            class="sc-bjUoiL KndDd"
          >
            https://test.com?q=
          </p>
          <table
            class="sc-crXcEl BVLoO"
          >
            <tbody>
              <tr
                class="sc-evZas caYqCy"
              >
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Search Provider
                </th>
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Prefix
                </th>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Allmusic
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /a
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Discogs
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /di
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Duck Duck Go
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /d
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <button
      class="sc-iBkjds dRHSwq"
      data-testid="toggle-button"
    >
      <i
        class="sc-ftvSup cWSRlf"
      >
        settings
      </i>
    </button>
  </div>
  <div>
    <div
      class="sc-papXJ dURgvU"
      hidden=""
    >
      <div
        class="sc-kDDrLX byPpqm"
      >
        <h2
          class="sc-dkzDqf jbDwiw"
        >
          Settings
        </h2>
        <button
          class="sc-iBkjds dRHSwq"
          data-testid="close-button"
        >
          <i
            class="sc-ftvSup cWSRlf"
          >
            close
          </i>
        </button>
      </div>
      <div
        class="sc-hHLeRK eelEQI"
      >
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Theme
          </h3>
          <div
            class="sc-iqcoie mUaeo"
          >
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Light
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-light"
              >
                <option
                  data-testid="option-light-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-light-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Dark
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-dark"
              >
                <option
                  data-testid="option-dark-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-dark-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-submit"
              >
                Apply
              </button>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-refresh"
              >
                Refresh
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="sc-iBkjds dRHSwq"
      data-testid="toggle-button"
    >
      <i
        class="sc-ftvSup cWSRlf"
      >
        settings
      </i>
    </button>
  </div>
</body>
`;

exports[`settings.tsx Tests settings rendering 3`] = `
<body>
  <div>
    <div
      class="sc-papXJ dURgvU"
      hidden=""
    >
      <div
        class="sc-kDDrLX byPpqm"
      >
        <h2
          class="sc-dkzDqf jbDwiw"
        >
          Settings
        </h2>
        <button
          class="sc-iBkjds dRHSwq"
          data-testid="close-button"
        >
          <i
            class="sc-ftvSup cWSRlf"
          >
            close
          </i>
        </button>
      </div>
      <div
        class="sc-hHLeRK eelEQI"
      >
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Theme
          </h3>
          <div
            class="sc-iqcoie mUaeo"
          >
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Light
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-light"
              >
                <option
                  data-testid="option-light-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-light-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Dark
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-dark"
              >
                <option
                  data-testid="option-dark-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-dark-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-submit"
              >
                Apply
              </button>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-refresh"
              >
                Refresh
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Search Providers
          </h3>
          <p
            class="sc-fEOsli OFSxW"
          >
            Default Search Provider
          </p>
          <p
            class="sc-bjUoiL KndDd"
          >
            https://test.com?q=
          </p>
          <table
            class="sc-crXcEl BVLoO"
          >
            <tbody>
              <tr
                class="sc-evZas caYqCy"
              >
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Search Provider
                </th>
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Prefix
                </th>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Allmusic
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /a
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Discogs
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /di
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Duck Duck Go
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /d
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <button
      class="sc-iBkjds dRHSwq"
      data-testid="toggle-button"
    >
      <i
        class="sc-ftvSup cWSRlf"
      >
        settings
      </i>
    </button>
  </div>
  <div>
    <div
      class="sc-papXJ dURgvU"
      hidden=""
    >
      <div
        class="sc-kDDrLX byPpqm"
      >
        <h2
          class="sc-dkzDqf jbDwiw"
        >
          Settings
        </h2>
        <button
          class="sc-iBkjds dRHSwq"
          data-testid="close-button"
        >
          <i
            class="sc-ftvSup cWSRlf"
          >
            close
          </i>
        </button>
      </div>
      <div
        class="sc-hHLeRK eelEQI"
      >
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Theme
          </h3>
          <div
            class="sc-iqcoie mUaeo"
          >
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Light
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-light"
              >
                <option
                  data-testid="option-light-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-light-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Dark
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-dark"
              >
                <option
                  data-testid="option-dark-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-dark-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-submit"
              >
                Apply
              </button>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-refresh"
              >
                Refresh
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="sc-iBkjds dRHSwq"
      data-testid="toggle-button"
    >
      <i
        class="sc-ftvSup cWSRlf"
      >
        settings
      </i>
    </button>
  </div>
  <div>
    <div
      class="sc-papXJ dURgvU"
      hidden=""
    >
      <div
        class="sc-kDDrLX byPpqm"
      >
        <h2
          class="sc-dkzDqf jbDwiw"
        >
          Settings
        </h2>
        <button
          class="sc-iBkjds dRHSwq"
          data-testid="close-button"
        >
          <i
            class="sc-ftvSup cWSRlf"
          >
            close
          </i>
        </button>
      </div>
      <div
        class="sc-hHLeRK eelEQI"
      >
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Search Providers
          </h3>
          <p
            class="sc-fEOsli OFSxW"
          >
            Default Search Provider
          </p>
          <p
            class="sc-bjUoiL KndDd"
          >
            https://test.com?q=
          </p>
          <table
            class="sc-crXcEl BVLoO"
          >
            <tbody>
              <tr
                class="sc-evZas caYqCy"
              >
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Search Provider
                </th>
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Prefix
                </th>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Allmusic
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /a
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Discogs
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /di
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Duck Duck Go
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /d
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <button
      class="sc-iBkjds dRHSwq"
      data-testid="toggle-button"
    >
      <i
        class="sc-ftvSup cWSRlf"
      >
        settings
      </i>
    </button>
  </div>
</body>
`;

exports[`settings.tsx Tests settings rendering 4`] = `
<body>
  <div>
    <div
      class="sc-papXJ dURgvU"
      hidden=""
    >
      <div
        class="sc-kDDrLX byPpqm"
      >
        <h2
          class="sc-dkzDqf jbDwiw"
        >
          Settings
        </h2>
        <button
          class="sc-iBkjds dRHSwq"
          data-testid="close-button"
        >
          <i
            class="sc-ftvSup cWSRlf"
          >
            close
          </i>
        </button>
      </div>
      <div
        class="sc-hHLeRK eelEQI"
      >
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Theme
          </h3>
          <div
            class="sc-iqcoie mUaeo"
          >
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Light
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-light"
              >
                <option
                  data-testid="option-light-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-light-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Dark
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-dark"
              >
                <option
                  data-testid="option-dark-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-dark-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-submit"
              >
                Apply
              </button>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-refresh"
              >
                Refresh
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Search Providers
          </h3>
          <p
            class="sc-fEOsli OFSxW"
          >
            Default Search Provider
          </p>
          <p
            class="sc-bjUoiL KndDd"
          >
            https://test.com?q=
          </p>
          <table
            class="sc-crXcEl BVLoO"
          >
            <tbody>
              <tr
                class="sc-evZas caYqCy"
              >
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Search Provider
                </th>
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Prefix
                </th>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Allmusic
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /a
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Discogs
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /di
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Duck Duck Go
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /d
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <button
      class="sc-iBkjds dRHSwq"
      data-testid="toggle-button"
    >
      <i
        class="sc-ftvSup cWSRlf"
      >
        settings
      </i>
    </button>
  </div>
  <div>
    <div
      class="sc-papXJ dURgvU"
      hidden=""
    >
      <div
        class="sc-kDDrLX byPpqm"
      >
        <h2
          class="sc-dkzDqf jbDwiw"
        >
          Settings
        </h2>
        <button
          class="sc-iBkjds dRHSwq"
          data-testid="close-button"
        >
          <i
            class="sc-ftvSup cWSRlf"
          >
            close
          </i>
        </button>
      </div>
      <div
        class="sc-hHLeRK eelEQI"
      >
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Theme
          </h3>
          <div
            class="sc-iqcoie mUaeo"
          >
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Light
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-light"
              >
                <option
                  data-testid="option-light-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-light-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <p
                class="sc-idiyUo fMJLXF"
              >
                Dark
              </p>
              <select
                class="sc-bczRLJ laxZOh sc-dIouRR bQrjcO"
                data-testid="select-dark"
              >
                <option
                  data-testid="option-dark-0"
                  value="0"
                >
                  Classic
                </option>
                <option
                  data-testid="option-dark-1"
                  value="1"
                >
                  Classic
                </option>
              </select>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-submit"
              >
                Apply
              </button>
            </div>
            <div>
              <button
                class="sc-gKXOVf foncKe"
                data-testid="button-refresh"
              >
                Refresh
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="sc-iBkjds dRHSwq"
      data-testid="toggle-button"
    >
      <i
        class="sc-ftvSup cWSRlf"
      >
        settings
      </i>
    </button>
  </div>
  <div>
    <div
      class="sc-papXJ dURgvU"
      hidden=""
    >
      <div
        class="sc-kDDrLX byPpqm"
      >
        <h2
          class="sc-dkzDqf jbDwiw"
        >
          Settings
        </h2>
        <button
          class="sc-iBkjds dRHSwq"
          data-testid="close-button"
        >
          <i
            class="sc-ftvSup cWSRlf"
          >
            close
          </i>
        </button>
      </div>
      <div
        class="sc-hHLeRK eelEQI"
      >
        <div
          class="sc-hAZoDl UZCWe"
        >
          <h3
            class="sc-hKMtZM sc-fnykZs ITLHV hkQEBd"
          >
            Search Providers
          </h3>
          <p
            class="sc-fEOsli OFSxW"
          >
            Default Search Provider
          </p>
          <p
            class="sc-bjUoiL KndDd"
          >
            https://test.com?q=
          </p>
          <table
            class="sc-crXcEl BVLoO"
          >
            <tbody>
              <tr
                class="sc-evZas caYqCy"
              >
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Search Provider
                </th>
                <th
                  class="sc-ksZaOG iVGklz"
                >
                  Prefix
                </th>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Allmusic
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /a
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Discogs
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /di
                </td>
              </tr>
              <tr
                class="sc-evZas caYqCy"
              >
                <td
                  class="sc-breuTD iQaTRx"
                >
                  Duck Duck Go
                </td>
                <td
                  class="sc-breuTD iQaTRx"
                >
                  /d
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <button
      class="sc-iBkjds dRHSwq"
      data-testid="toggle-button"
    >
      <i
        class="sc-ftvSup cWSRlf"
      >
        settings
      </i>
    </button>
  </div>
  <div />
</body>
`;

exports[`settings.tsx Tests tables 1`] = `
<body>
  <div>
    <table
      class="sc-crXcEl BVLoO"
    >
      <tbody>
        <tr
          class="sc-evZas caYqCy"
        >
          <th
            class="sc-ksZaOG iVGklz"
          >
            Test
          </th>
        </tr>
        <tr
          class="sc-evZas caYqCy"
        >
          <td
            class="sc-breuTD iQaTRx"
          >
            Test
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;
